### ISSUE TITLE
Handling `unicode` in recurly configuration

### ISSUE THREAD

**Issue by maximehardy** [recurly/recurly-client-python#138]
I was trying to create accounts from the API of Recurly using the Python client. Python 2.7, recurly 2.2.17

When creating a new account with a first name (or any other attribute actually) containing non-ascii characters (fetched as a `unicode`), saving the entity raises `UnicodeDecodeError: 'ascii' codec can't decode byte [...]`

A simple `account.first_name = u'Frédérique-Fançois'.encode('utf-8')` still raises the same error but at another level in the module.

For more details about the problem see this [stackoverflow post](http://stackoverflow.com/questions/33783783/recurly-python-client-raising-unicodedecodeerror-when-supplying-non-ascii-charac).

This gist reproduces the problem and explains the details of the problem: [code](https://gist.github.com/maximehardy/d3a0a6427d2b6791b3dc)

To sum up the solution is to ensure that parameters like `recurly.API_KEY` come into `httplib` headers as `str`.

I think that documentation about the issue would be a good start, but going further you could either raise an error if configuration values are `unicode` or you could force a conversion (but be careful with encoding).

**Comment by bhelx** [comment]
Hey @maximehardy

Thanks for the detailed report! I'm thinking what might work is to throw an error somewhere around here if a unicode string is detected for the api key or the subdomain.

https://github.com/recurly/recurly-client-python/blob/157b25be7ccab396181c2e26da2149695c030f22/recurly/resource.py#L283

I'll probably also update the readme. Does that sound reasonable?

**Comment by maximehardy** [comment]
Sounds good considering the current architecture.

**Comment by bhelx** [comment]
Closed due to https://github.com/recurly/recurly-client-python/pull/141
